compiler/rustc_ty_utils/src/layout.rs

```diff
@@ -277,7 +266,7 @@ fn layout_of_uncached<'tcx>(
         ty::Closure(_, ref substs) => {
             let tys = substs.as_closure().upvar_tys();
             univariant(
-                &tys.map(|ty| Ok(cx.layout_of(ty)?.layout)).collect::<Result<Vec<_>, _>>()?,
+                &tys.map(|ty| Ok(cx.layout_of(ty)?.layout)).try_collect::<IndexVec<_, _>>()?,
```

Comment:
at with that you don't need the turbofish anymore at all, right?

Reply:
I do, because the parameter to `univariant` is `&IndexSlice`, so if I don't turbo-fish it it tries to collect into an `IndexSlice`, which of course doesn't work.